### PR TITLE
Remove quantum solar panels

### PR DESCRIPTION
**Pull request by ymber**
#### Summary
SUMMARY: Content "Remove quantum solar panels"

#### Purpose of change
Quantum solar panels are well above the efficiency of existing solar panels and aren't well justified by current lore.

#### Describe the solution
Remove quantum solar panels

#### Describe alternatives you've considered
Aftershock might want these.

#### Testing
Game loads as expected.

### CODE REVIEW

**Comment by davidpwbrown** [comment]
Will aftershock take them on do you think? @Maleclypse

**Comment by Kelenius** [comment]
They should probably be replaced by something in the lab finales.

**Comment by eilaattwood** [comment]
> They should probably be replaced by something in the lab finales.

The best candidate is a vibrator, in my opinion.

**Comment by SirPendrak** [comment]
> They should probably be replaced by something in the lab finales.

I propose a stack of ESAPI ballistic vests

**Comment by I-am-Erk** [comment]
Given that I've been slowly removing plutonium fuel cells from various places, but they're still something we want in game, the quantum solar lab finale could be a locker full of fuel cells instead

**Comment by Brian-Otten** [comment]
The plutonium fuel cell stash would be a great lab finale, as a replacement for the solar panels or otherwise.

**Comment by ymber** [comment]
Added plutonium cells where quantum solar panel lab spawns used to be.

**Comment by Brian-Otten** [comment]
I think 1-10 plutonium cells might feel a bit disappointing as a lab finale reward, specifically the chance for it to be only 1 or 2. It seems that if it is the labs plutonium cell storage they would have a good amount, and it being weighted something more like 8-20 would make it feel more on par with the other lab finales balance wise. if we feel that's a bit much making it at least 3-10 would also work.

**Comment by Maleclypse** [comment]
> Will aftershock take them on do you think? @Maleclypse

Yes.  It will probably go well with the new supertech crafting.  We won't be able to put it back into lab finale's until those are JSONized, right?

**Comment by anothersimulacrum** [comment]
Nope. It'd be fairly easy to JSON-ize those though (verging on JSON-only, just need someone to rip out the old code).

**Comment by I-am-Erk** [comment]
Since I'm trying to make JSON lab maps, I'd appreciate that

**Comment by Maleclypse** [comment]
I see the mapgen.cpp won't really affect Aftershock taking on these two items but will the bionics.cpp, iuse.cpp and items.cpp changes prevent these items from working in Aftershock?

**Comment by ymber** [comment]
Yeah that code was supporting the quantum solar backpack. Hardcoding item ID strings was a bad way of implementing it and it needs generalizing.

File: src/mapgen.cpp [hunk 1]
```diff
@@ -4175,14 +4175,14 @@ void map::draw_lab( mapgendata &dat )
                             spawn_item( point( SEEX, SEEY ), "plasma_gun" );
                             spawn_item( point( SEEX, SEEY ), "plasma" );
                             spawn_item( point( SEEX - 1, SEEY ), "recipe_atomic_battery" );
-                            spawn_item( point( SEEX, SEEY  - 1 ), "solar_panel_v3" );
+                            spawn_item( point( SEEX + 1, SEEY ), "plut_cell", rng(1, 10) );
```
Comment (kevingranade) [inline src/mapgen.cpp]:
```suggestion
                            spawn_item( point( SEEX + 1, SEEY ), "plut_cell", rng( 1, 10 ) );
```

File: src/mapgen.cpp [hunk 2]
```diff
@@ -4175,14 +4175,14 @@ void map::draw_lab( mapgendata &dat )
...
                         } else if( loot_variant > 67 && loot_variant < 89 ) {
                             spawn_item( point( SEEX - 1, SEEY - 1 ), "mininuke", dice( 3, 6 ) );
                             spawn_item( point( SEEX, SEEY - 1 ), "mininuke", dice( 3, 6 ) );
                             spawn_item( point( SEEX - 1, SEEY ), "mininuke", dice( 3, 6 ) );
                             spawn_item( point( SEEX, SEEY ), "mininuke", dice( 3, 6 ) );
                             spawn_item( point( SEEX, SEEY ), "recipe_atomic_battery" );
-                            spawn_item( point( SEEX, SEEY ), "solar_panel_v3" );
+                            spawn_item( point( SEEX + 1, SEEY ), "plut_cell", rng(1, 10) );
```
Comment (kevingranade) [inline src/mapgen.cpp]:
```suggestion
                            spawn_item( point( SEEX + 1, SEEY ), "plut_cell", rng( 1, 10 ) );
```

File: src/mapgen.cpp [hunk 3]
```diff
@@ -4213,7 +4213,7 @@ void map::draw_lab( mapgendata &dat )
                                      point( SEEX + 1, SEEY - 1 ), false, calendar::start_of_cataclysm );
                         place_items( "guns_rare", 96, point( SEEX - 2, SEEY ), point( SEEX + 1, SEEY ), false,
                                      calendar::start_of_cataclysm );
-                        spawn_item( point( SEEX + 1, SEEY ), "solar_panel_v3" );
+                        spawn_item( point( SEEX + 1, SEEY ), "plut_cell", rng(1, 10) );
```
Comment (kevingranade) [inline src/mapgen.cpp]:
```suggestion
                        spawn_item( point( SEEX + 1, SEEY ), "plut_cell", rng( 1, 10 ) );
```

**Comment by kevingranade** [comment]
If you're going to nuke the support code, you need to migrate the quantum panels instead of just obsoleting them.  They're still going to exist, but this PR makes them silently stop working.

So either leave in the code that makes them work or add migration entries to turn existing quantum panels into some kind of remaining solar panel.

**Comment by Maleclypse** [comment]
Does that mean I was right to ask above if this current PR would prevent them from working in Aftershock?

**Comment by esotericist** [comment]
> Does that mean I was right to ask above if this current PR would prevent them from working in Aftershock?

yeah, this PR as written will make quantum solar backpacks non-functional.

given that there's desire to keep the item extant in aftershock, what should probably happen instead is adding a mechanism for a backpack to specify its efficiency.

**Comment by ymber** [comment]
Yeah specifying power output for generator type items was the plan from the start. I just haven't had time to sort this out properly.

**Comment by ZhilkinSerg** [comment]
Can you resolve conflicts please?

**Pull request merged**